nni/compression/pytorch/quantization_speedup/integrated_tensorrt.py

```diff
@@ -326,6 +329,8 @@ def inference(self, test_data):
             Model input tensor
         """
         # convert pytorch tensor to numpy darray
+        if test_data.device != torch.device("cpu"):
+            test_data = test_data.to("cpu")
```

Comment:
it is a little strange that inference can only be executed on cpu... why?

Reply:
![image](https://user-images.githubusercontent.com/50486294/122752152-2966f100-d2c3-11eb-9779-27dd4146b42b.png)
Actually the inference executed on GPU after the memory copy from host memory to device memory which is done in our tool and the process of host2device memory copy is unavoidable. We use pycuda' s `memcpy_htod_async` api to do such copy. It is just a way of unifying that we choose to handle all host data and we will do memory copy in our engine before calibration and inference. We can also take another to handle it such as finding other way or API which can use torch cuda data directly. However, it will be more complicated and no extra advantage.

Reply:
ok, got it